File: tourneyAPI/Models/Entities/Characters/Wario.cs
namespace Entities;

using Enums;

// Defines the competitive profile metadata for this playable character.
public class Wario : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public Wario()
    {
        Id = CharacterId.Wario;
        characterName = CharacterName.WARIO;
        archetype = Archetype.HIT_AND_RUN;
        fallSpeed = FallSpeed.FAST_FALLERS;
        weightClass = WeightClass.MIDDLEWEIGHT;
        tierPlacement = TierPlacement.A;
    }

}

File: tourneyClient/src/models/entities/Characters/Wario.ts
import { CharacterId } from "@/models/Enums/CharacterId";
import { Character } from "../Character"
import { CharacterName } from "@/models/Enums/CharacterName";
import { Archetype } from "@/models/Enums/Archetype";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { WeightClass } from "@/models/Enums/WeightClass";
import { TierPlacement } from "@/models/Enums/TierPlacement";

export const Wario: Character =
{
    id: CharacterId.Wario,
    characterName: CharacterName.WARIO,
    archetype: Archetype.HIT_AND_RUN,
    fallSpeed: FallSpeed.FAST_FALLERS,
    weightClass: WeightClass.MIDDLEWEIGHT,
    tierPlacement: TierPlacement.A
}

export default Wario; 